src/react/boosts/BoostCard.tsx

```diff
@@ -19,94 +21,134 @@ export interface Items {
 }
 
 export interface BoostCardProps {
+  channelId: string;
   open: boolean;
   boostId: string;
   title: string;
   goalAmount: number;
   paymentsTotal: number;
+  dateEnd: string;
   why: string;
   what: string;
   about: string;
   items: Items[];
-  handlePayments(): void;
+  handlePayments(newsroom: string): void;
 }
 
-// TODO(sruddy) get newsroom data from address
+// TODO(sruddy) get newsroom image
 const boost = {
   image: "https://cdn.mos.cms.futurecdn.net/ewcvC8bNBec6oMG9zufgVg.jpg",
-  newsroom: "Block Club Chicago",
-  newsroomUrl: "https://blockclubchicago.org/",
-  newsroomRegistryUrl: "https://registry.civil.co/listing/0x23daa7fba48cd68a2b86a77a1e707a6aae41c4ea",
-  daysLeft: 15,
 };
 
-export const BoostCard: React.FunctionComponent<BoostCardProps> = props => {
-  return (
-    <BoostWrapper open={props.open}>
-      <BoostImg>
-        <img src={boost.image} />
-      </BoostImg>
-      <BoostTitle>
-        {props.open ? <>{props.title}</> : <a href={"/boosts/" + props.boostId}>{props.title}</a>}
-      </BoostTitle>
-      <BoostNewsroomInfo>
-        <BoostNewsroom>{boost.newsroom}</BoostNewsroom>
-        {props.open && (
+export class BoostCard extends React.Component<BoostCardProps> {
+
+  public render(): JSX.Element {
+    const daysLeft = this.daysLeft(this.props.dateEnd);
+    const addr = this.props.channelId;
+    let newsroomName = "";
+
+    return (
+      <BoostWrapper open={this.props.open}>
+        <BoostTitle>
+          {this.props.open ? <>{this.props.title}</> : <a href={"/boosts/" + this.props.boostId}>{this.props.title}</a>}
+        </BoostTitle>
+        <BoostImg>
+          <img src={boost.image} />
+        </BoostImg>
+
+        <Query query={boostNewsroomQuery} variables={{ addr }}>
+          {({ loading, error, data }) => {
+            if (loading) {
+              return <></>;
+            } else if (error) {
+              console.log(JSON.stringify(error));
+              return <></>;
+            }
+
+            if (data.listing !== null) {
+              newsroomName = data.listing.name;
+              return (
+                <BoostNewsroomInfo>
+                  <BoostNewsroom>{newsroomName}</BoostNewsroom>
+                  {this.props.open && (
+                    <>
+                      <a href={data.listing.url} target="_blank">
```

Comment:
At first I thought `listing.url` would be the registry listing not the newsroom listing, but I just checked graphql on both staging and production and found `listing.url` to be blank, unless I'm doing something wrong? I think the newsroom URL is only in the charter.

Reply:
@am9u said that this was the newsroom url not registry. I still have some work to get the image so if this is blank I will add this also.